Add the meeting minutes for 2022-12-06 (#26)
ocapn/ocapn: meeting-minutes/2022-12-06.md

warner: I'm Brian Warner, I've breen doing several captp things, built one in Python, in Javascript for Agoric I made one called swingset, dont' have handoffs but we do have promise pipelining

DanC: Dan Connolly, I'm very interested in spritely <-> agoric interop, I used to do w3c standards stuff, now doing ocaps at agoric

Dmitri: Hi I'm Dmitri Zagidulin, I work on VCs and DIDs, but also helped with zcap specification / zcap-ld libraries, so I'm a huge fan of cert style ocaps, I'm interestd in ocapn, couldn't quite understand it, but christine said ocapn rules so I want to join and see if I can understand it better

JakobWeisz: I'm a community contributor to SandstormIO, I found I was basically complaining on G+ at the time about the awful state of cloud stuff and wanted to take my apps with me, kentonv said hey I'm working on something, so I've been following sandstorm since there was such a thing.  it takes webapps and lets you easily run them from one platform and ocap security is built in at a core level with capn proto

> isd: I'm Ian Denhart, also a sandstorm contributor, also way back wrote a handful of plugins for gnu social that never got merged, but sort of interesting thing from this group is the sandstorm stuff and I'm also maintainer of the Go/Haskell versions of cap'n proto

Reviewer: ```suggestion
isd: I'm Ian Denhardt, also a sandstorm contributor, also way back wrote a handful of plugins for gnu social that never got merged, but sort of interesting thing from this group is the sandstorm stuff and I'm also maintainer of the Go/Haskell versions of cap'n proto
```

Author: Fixed, thanks!

isd: Jakob alluded to this but it's roughly a captp

isd: I think with that, spritely, agoric, those are the 3 we're looking to reconcile

danc: when I said spritely <-> agoric interop, I should have included cap'n proto.

elplatt: I'm Edward Platt, relatively new to ocap, largely came to it through following spritely, really excited about it, really fits well with many of the things I've been thinking about, really looking forward to being part of the future, learn from all of y'all

jar286: I'm Jonathan Rees, I worked on lisp implementation and standardization back in the 80s, and lisp/scheme in the 90s, and ocap was in parallel, in ignorance with Joule/E, so that was amazing to discover it was all happening at the time.  So I then discovered all the huge problems I cared about were solved, so I went to work at millenial pharmaceuticals, Science Commons, Open Tree of LIfe, and reconnected with Christine since then,

> jar286: got re-interested in ocaps.  And yes I was on the W3C TAG for 4 years or so together

Reviewer 2: Should be 'on the W3C TAG with Dan C for 4 years'

(because I had said this in response to a prompt from Christine asking me to make the Dan C connection)

Reviewer 2: (and because the upcoming "we" refers to JAR and Dan C, not JAR and Christine)

Author: Fixed, thanks!

Reviewer 3: I expect there's a re-review button next to @jar398 's name above; hit that to prompt him to remove his request for changes?

Or @jar398 , go ahead and signal whether the edits satisfy your request?

markm: think that the most important starting point to contribute is that the interop should be focused on the abstract syntax not the concrete syntax.  the abstract syntax is what we must all agree on, if we can agree on that we can do adaptors for the concrete syntax

Jessica: sounds great, hope we can agree on concrete syntax too but even that is good

mathieu: Mathieu Hofman, I work at Agoric with Waner, MarkM, etc.  before at agoric was trying to build on top of basic RPCs, so I've always been interested in that.  I ended up reinventing many distributed GCs ideas that MarkM and others invented 20+ years ago

mathieu: interested in how we can converge

mikayla: I'm Mikayla Maki, I'm kind of just hanging around to keep an eye on things.  been getting into ocaps, recently did the heart of spritely, asynchronously now, hoping to expand that to actually hosting and serving my personal blog

juliana: Hi, I'm Juliana, I'm just kind of here to listen honestly. I heard about OCap through Spritely and am really excited for its potential to make distributed computing more accessible and secure.

> MikeStay: I've worked with MarkM at Google on the Caja project, right now I'm the CTO of Pyrofex -  corporation working on a new blockchain based around Go, smart contracts tend to be communicating over OCapN, so would like to be part of this

Author: Fixed, thanks!

danc: there are some smaller vattps

markm: we do have a p2p protocol, and both vats talking to kernel, and one chain talking to other chains or non-chains.  we don't have chains talking to other chains yet

markm: there are some hard problems to take on, one of the constraints of having an endpoint be a conventional blockchain is that a computationally conventional blockchain cannot keep secrets, so that has constrained the captp architecture in particular.  helps bootstrap from offline to online connectivity.  in certain conventions impossible to do a captp as a bearer right (sturdyrefs).  the ordering issues are important and probably

markm: things we're going to need a variety, not a one-size-fits-all, and the thing we'll probably repeatedly revisit is distributed failure models... to what degree are failures masked or apparent for means for participants to recover from various failures

Jessica: ian would you like to present on capn proto?

> isd: yes one difference is that we have substantial existing users and deployments so unfortunately we have a lot less flexibility in terms of change than others, but we currently have no implementation that does 3 party handoffs, there's some datatypes and specs in captp.capnp but that's likely to change in next few months.  have a client that wants various improvements in go handoffs

Reviewer: ```suggestion
isd: yes one difference is that we have substantial existing users and deployments so unfortunately we have a lot less flexibility in terms of change than others, but we currently have no implementation that does 3 party handoffs, there's some datatypes and specs in rpc.capnp but that's likely to change in next few months.  have a client that wants various improvements in go, including handoffs
```

isd: so that's interesting to me to make sure that we have better chance of interop

isd: so I'd like to see if we have a way to shake out requirements early, so that's what's kind of going on on the go side

> isd: the other interesting thing within last week or so is that one of the forgefed people (forgefed attempting to do federated code forges) was planning on doing stuff with activitypub but has found that it's annoying to do anything but social pubsub, but they're interested, we wanted to possibly pull them in but they're in israel/palestine at the moment

Reviewer: What got lost here is that Pere is actually prototyping stuff with haskell-capnp

Reviewer 3: @zenhack maybe format your comment using the +/- diff thingy in such a way that @tsyesika can accept it with 1 button?

> isd: the other interesting thing within last week or so is that one of the forgefed people (forgefed attempting to do federated code forges) was planning on doing stuff with activitypub but has found that it's annoying to do anything but social pubsub, but they're interested, we wanted to possibly pull them in but they're in israel/palestine at the moment

Reviewer: ```suggestion
isd: the other interesting thing within last week or so is that one of the forgefed people (forgefed attempting to do federated code forges) was planning on doing stuff with activitypub but has found that it's annoying to do anything but social pubsub, but they're interested in captp and prototyping things with haskell-capnp, we wanted to possibly pull them in but they're in israel/palestine at the moment
```
